libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1066,11 +1067,18 @@ void GCS_MAVLINK::send_scaled_pressure(AP_Baro &barometer)
 {
     uint32_t now = AP_HAL::millis();
     float pressure = barometer.get_pressure(0);
+    float diff_pressure = 0; // pascal
+
+    AP_Airspeed *airspeed = AP_Airspeed::get_singleton();
+    if (airspeed != nullptr) {
```

Comment:
this kind of breaks the model for singletons, but it certainly gets the job done nicely without defines. Could we at least squeeze in a comment that says "This singleton will be null when the vehicle doesn't load the library such as in copter"?

I'm thinking we should make this type of change in as many places possible, maybe this is a good way to get all the copter builds into a single binary?

Reply:
> this kind of breaks the model for singletons

It certainly doesn't break the model because every singleton has a getter. And two of the oldest singletons, DataFlash and AP_InertialSensor, work this way (although the latter recently got the new AP method returning a reference).

> Could we at least squeeze in a comment that says "This singleton will be null when the vehicle doesn't load the library such as in copter"?

I see zero need for that, it's completely clear why it would be null.

> I'm thinking we should make this type of change in as many places possible, maybe this is a good way to get all the copter builds into a single binary?

That makes the code very verbose, the reason to have the AP methods return a reference is precisely so that we don't have to check for null. For some libraries that's fine as they aren't going to be removed in a build, for others it can be a problem.
And this won't help with Copter builds.

Reply:
I misunderstood the deeper meaning of singletons. I always thought that singletons could not have multiple instances and were guaranteed to be non-null.. but after a little research I retract my concern.

Reply:
Voting for AP method!